go/vt/mysqlctl/proto/schema_test.go

```diff
@@ -177,3 +177,115 @@ func TestSchemaDiff(t *testing.T) {
 	sd2.TableDefinitions = append(sd2.TableDefinitions, &TableDefinition{Name: "table2", Schema: "schema3", Type: TABLE_BASE_TABLE})
 	testDiff(t, sd1, sd2, "sd1", "sd2", []string{"sd1 and sd2 disagree on schema for table table2:\nschema2\n differs from:\nschema3"})
 }
+
+func TestFilterTables(t *testing.T) {
+	var testcases = []struct {
+		desc          string
+		input         *SchemaDefinition
+		tables        []string
+		excludeTables []string
+		includeViews  bool
+		want          *SchemaDefinition
+	}{
+		{
+			desc: "filter based on tables (whitelist)",
+			input: &SchemaDefinition{
+				DatabaseSchema: "CREATE DATABASE {{.DatabaseName}}",
```

Comment:
I believe you can drop the DatabaseSchema field for the SchemaDefinitions being used here; they shouldn't be needed for this test.